www/guides/react.md

```diff
@@ -0,0 +1,361 @@
+
+Create a new file at `src/App.jsx` and paste the following code into this new file to create an `App` component:
+
+```jsx
+import React, {useState, useEffect} from 'react';
+
+function App() {
+  // Create the count state.
+  const [count, setCount] = useState(0);
+  // Create the counter (+1 every second).
+  useEffect(() => {
+    const timer = setTimeout(() => setCount(count + 1), 1000);
+    return () => clearTimeout(timer);
+  }, [count, setCount]);
+  // Return the App component.
+  return (
+    <div className="App">
+      <header className="App-header">
+        <p>
+          Edit <code>src/App.jsx</code> and save to reload.
+        </p>
+        <p>
+          Page has been open for <code>{count}</code> seconds.
+        </p>
+        <p>
+          <a
+            className="App-link"
+            href="https://reactjs.org"
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            Learn React
+          </a>
+        </p>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+Now include it in `index.jsx`
+
+```diff
+import React from 'react';
+import ReactDOM from 'react-dom';
++ import App from './App.jsx';
+
+- ReactDOM.render(<div>"HELLO WORLD"</div>, document.getElementById('root'));
++ ReactDOM.render(
++    <React.StrictMode>
++      <App />
++    </React.StrictMode>,
++    document.getElementById('root'),
++  );
+```
+
+> 💡 Tip: [Strict Mode](https://reactjs.org/docs/strict-mode.html) is a tool for highlighting potential problems in React code.
+
+You shouldn't need to restart Snowpack to see this, it should look like this:
+
+Image: screenshot of the project, showing text saying "edit `src/App.jsx` and save to reload," text that says "Page has been open for" and the number of seconds then "seconds". A link with the text Learn React that links to the React website.
+
+## Styling your Snowpack/React project
+
+When you add assets like images or CSS, Snowpack includes them in your final build. If you already know React, this process should look pretty familiar.
+
+> 💡 Tip: As you're doing this, you should not need to reload the page or restart Snowpack. Snowpack automatically updates the project in the browser as you edit code.
+
+Add this file `logo.svg` (link to file) to your `src` directory. Now you can import it into your `App.jsx` and use it in an `img` tag to display it.
+
+```diff
+import React, { useState, useEffect } from 'react';
++ import logo from './logo.svg';
+
+function App() {
+  // Create the count state.
+  const [count, setCount] = useState(0);
+  // Create the counter (+1 every second).
+  useEffect(() => {
+    const timer = setTimeout(() => setCount(count + 1), 1000);
+    return () => clearTimeout(timer);
+  }, [count, setCount]);
+  // Return the App component.
+  return (
+    <div className="App">
+      <header className="App-header">
++        <img src={logo} className="App-logo" alt="logo" />
```

Comment:
can you add the 1 line below this one, so that people get a better idea of where this should go in their file?